main/lsp/LSPTypechecker.cc

```diff
@@ -415,21 +416,11 @@ bool LSPTypechecker::runSlowPath(LSPFileUpdates updates, WorkerPool &workers, bo
     return committed;
 }
 
-void LSPTypechecker::pushDiagnostics(u4 epoch, vector<core::FileRef> filesTypechecked,
-                                     vector<std::unique_ptr<core::Error>> errors) {
+void LSPTypechecker::pushAllDiagnostics(u4 epoch, vector<core::FileRef> filesTypechecked,
+                                        vector<std::unique_ptr<core::Error>> errors) {
     config->logger->debug("[Typechecker] Sending diagnostics for epoch {}", epoch);
-    vector<core::FileRef> errorFilesInNewRun;
     UnorderedMap<core::FileRef, vector<std::unique_ptr<core::Error>>> errorsAccumulated;
-
-    // Update epochs of all files that were typechecked, since we've recalculated the set of diagnostics in these files.
-    for (auto f : filesTypechecked) {
-        // N.B.: Overflow could theoretically happen. It would take an absurdly long time for someone to make
-        // 4294967295 edits in one session. One way to handle that case: Have a special overflow request that blocks
-        // preemption and resets all versions to 0.
-        if (diagnosticEpochs[f.id()] < epoch) {
-            diagnosticEpochs[f.id()] = epoch;
-        }
-    }
+    ErrorReporter er(move(config));
```

Comment:
`er` should be a field on `LSPTypechecker`. It stores state that needs to persist across diagnostic pushes. With this code, LSPTypechecker will "forget" about which files have errors after every typecheck.